app/models/test.rb

```diff
@@ -3,11 +3,22 @@ class Test < ApplicationRecord
   belongs_to :author, class_name: "User", foreign_key: "author_id", optional: true
   has_many :questions, dependent: :destroy
   has_many :tests_users
-  has_many :users, through: :tests_users
+  has_many :users, through: :tests_users, dependent: :destroy
 
-  def self.find_by_category_name(category_name)
-    joins(:category).where(categories: {title: category_name})
+  scope :easy, -> { where(level: TestGuru::EASY_LEVEL) }
+  scope :medium, -> { where(level: TestGuru::MEDIUM_LEVEL) }
+  scope :hard, -> { where(level: TestGuru::HARD_LEVEL) }
+  scope :by_level, -> (level) { where(level: level) }
+
+  scope :find_by_category_name, ->(category_name) {
+    joins(:category).where(categories: { title: category_name })
                     .order(title: :desc)
-                    .pluck(:title)
+  }
+
+  validates :level, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :title, presence: true, uniqueness: { scope: :level }
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

про правильную миграцию можно посмотреть к примеру на [СО](https://stackoverflow.com/questions/32927262/create-migration-with-uniqueness-to-a-scope)